docs/tools/near-indexer.md

```diff
@@ -0,0 +1,487 @@
+
+---
+
+## Setup
+
+### Requirements
+
+- [Rust.](https://www.rust-lang.org/) If not already installed, please [follow these instructions](https://docs.near.org/docs/tutorials/contracts/intro-to-rust#3-step-rust-installation).
+- Minimunm hardware:
+  - 4 CPU cores
+  - 16GB RAM
+  - 100GB SSD _(HDD will **not** work)_
+
+### Creating your project
+
+To create a new project with Rust, you will start by creating a new binary by running the following command in your terminal:
+
+```bash
+cargo new --bin example-indexer
+```
+
+Now change directory into your newly created project:
+
+```bash
+cd example-indexer
+```
+
+Inside this folder you will find:
+
+- `cargo.toml`
+- `src` folder with a `main.rs` file inside
+
+### Create Rust Toolchain
+
+Next, you will need to create a Rust toolchain that mirrors the one in [`nearcore`](https://github.com/near/nearcore/blob/master/rust-toolchain):
+
+To do this, run the following command in the root of your project: _(be sure to check the link above for the correct `nightly` version)_
+
+```bash
+echo nightly-2020-10-08 > rust-toolchain
+```
+
+### Add dependencies
+
+**1) In your `cargo.toml` file add `near-indexer` under [dependencies]:**
+
+```toml
+[dependencies]
+near-indexer = { git = "https://github.com/near/nearcore" }
+```
+
+**Note:** While it is fine to omit specific commit hash for this tutorial we highly recommend to freeze near-indexer dependency for specific commit from the `nearcore` repository. _(Example below)_
+
+```toml
+near-indexer = { git = "https://github.com/nearprotocol/nearcore", rev="29fcaf3b8c81a4c0371d105054ce251355382a77" }
+```
+
+**2) Add `actix`, `openssl-probe`, & `tokio`**
+
+```toml
+actix = "0.11.0-beta.1"
+openssl-probe = { version = "0.1.2" }
+tokio = { version = "1.1", features = ["sync"] }
+```
+
+**3) Once complete, your `cargo.toml` dependencies should look something like this:**
+
+```toml
+[dependencies]
+near-indexer = { git = "https://github.com/near/nearcore" }
+actix = "0.11.0-beta.1"
+openssl-probe = { version = "0.1.2" }
+tokio = { version = "1.1", features = ["sync"] }
+```
+
+**4) Install and check dependencies**
+
+Install and check dependencies by running:
+
+```bash
+cargo check
+```
+
+<blockquote class="warning">
+<strong>heads up</strong><br><br>
+
+If the cargo check command fails with some errors it might be because of different versions of underlying dependencies. 
+
+- A quick solution is to copy `Cargo.lock` from `nearcore` repository [ [here](https://raw.githubusercontent.com/near/nearcore/master/Cargo.lock) ]  and replace it with the contents of your project's `Cargo.lock` file.
+- After this is complete, rerun `cargo check` to see if this resolves your errors.
+
+</blockquote>
+
+---
+
+## Constructing `main.rs`
+
+> Now that we have our basic setup, we need to update `main.rs`.  
```

Comment:
Is there a reason why this paragraph is wrapper in pretty yellow block?

Reply:
It's a stylistic preference.  The beginning of each new section starts with a highlighted intro.